src/lightning_app/storage/path.py

```diff
@@ -385,6 +385,10 @@ def shared_storage_path() -> pathlib.Path:
     ``SHARED_MOUNT_DIRECTORY`` environment variable. In the cloud, the shared path will point to a S3 bucket. All Works
     have access to this shared dropbox.
     """
+    storage_path = os.getenv("LIGHTNING_STORAGE_PATH", "")
+    if storage_path != "":
+        return pathlib.Path(storage_path)
+
     bucket_name = os.getenv("LIGHTNING_BUCKET_NAME", "")
     app_id = os.getenv("LIGHTNING_CLOUD_APP_ID", "")
```

Comment:
After the change now, when will this code path be triggered?

Maybe what that means is that from now on, `LIGHTNING_STORAGE_PATH` has the bucket name and app id already combined. In this case, I'm wondering why this code would still be needed? Maybe add a TODO here that once the transitioning phase is over, we want to remove it.

Reply:
Yes, a good point!
I'll add the test and the comment. We assume `LIGHTNING_STORAGE_PATH` will be passed from BE side, but we'd like to leave the old way for some time for compatibility reasons.

Reply:
We assume this will be activated by the BE release with corresponding BE change (setting the envvar to the container). In practice, we can expect it on Sept. 19-21th.

Reply:
Yes, let's add a TODO to remove the previous logic